dataloader/dataloader.hpp

```diff
@@ -0,0 +1,215 @@
+/**
+ * @file dataloader.hpp
+ * @author Kartik Dutt
+ * 
+ * Definition of Dataloader for popular datasets.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/core/data/scaler_methods/min_max_scaler.hpp>
+#include <mlpack/core/math/shuffle_data.hpp>
+#include <mlpack/core/data/split_data.hpp>
+#include <mlpack/prereqs.hpp>
+#include <dataloader/datasets.hpp>
+#include <mlpack/core.hpp>
+#include <utils/utils.hpp>
+
+
+#ifndef MODELS_DATALOADER_HPP
+#define MODELS_DATALOADER_HPP
+
+template<
+  typename DatasetX = arma::mat,
+  typename DatasetY = arma::mat,
+  class ScalerType = mlpack::data::MinMaxScaler
+>
+class DataLoader
+{
+ public:
+  //! Create DataLoader object.
+  DataLoader();
+
+  /**
+   * Constructor for DataLoader. This is used for loading popular Datasets such as
+   * MNIST, ImageNet, Pascal VOC and many more.
+   * 
+   * @param datasetPath Path or name of dataset.
+   * @param shuffle whether or not to shuffle the data.
+   * @param ratio Ratio for train-test split.
+   * @param augmentation Adds augmentation to training data only.
+   * @param augmentationProbability Probability of applying augmentation on dataset.
+   */
+  DataLoader(const std::string &dataset,
```

Comment:
Looks like the comments don't match with the parameters, but maybe this isn't ready yet?

Reply:
> Looks like the comments don't match with the parameters

Terribly sorry about, the dataloader went through different versions to remove redundant / duplicate code and I think that led to incomplete parameters in it.

> but maybe this isn't ready yet?

There are only few points left. I was writing more tests for models, utils and dataloader. In the next few hours I will remove all dependencies and switch to boost lib. I will review my PR and make the required style changes like this one. Maybe you could take a look after that if you get a chance. Thanks a lot.